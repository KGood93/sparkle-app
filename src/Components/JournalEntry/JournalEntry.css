.journalEntry {
    border: solid rgb(7,123,136) 3px;
    background-color: rgb(241,221,210);
    margin-top: 5px;
    width: 250px;
}

.journalEntry a {
    color: rgb(7,123,136);
    font-family: 'Playfair Display', serif;
    font-weight: bold;
    font-size: 25px;
    margin-left: 5px;
}

.journalEntry a:hover {
    color: rgb(251,159,139);
}

.journalEntry_delete {
    font-family: 'Playfair Display', serif;
    font-weight: bold;
    font-size: 15px;
    color: rgb(241,221,210);
    display: block;
    margin-bottom: 5px;
    margin-right: auto;
    margin-left: auto;
    background-color: rgb(7,123,136);
    border: solid 2px rgb(107,182,188);
    padding: 3px 8px 3px 8px;
}

.date {
    font-family: 'Playfair Display', serif;
    font-size: 14pt;
    margin: 5px;
}

@media screen and (min-width: 450px) {
    .journalEntry {
        width: 300px;
    }
}