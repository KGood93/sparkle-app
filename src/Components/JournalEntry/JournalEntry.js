import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import {allEntries} from '../../STORE'
import './JournalEntry.css'
import ApiContext from '../../ApiContext'
import config from '../../config'

//const entries = allEntries().entries
//console.log(entries)

class JournalEntry extends Component {
    static contextType = ApiContext

    handleClickDelete = e => {
        e.preventDefault()
        const entryid = e.target.id
        console.log(entryid)

        fetch(`${config.API_ENDPOINT}/entry/${entryid}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json'
            },
        })
            .then(res => {
                if(!res.ok)
                    return res.json().then(e => Promise.reject(e))
                    return res.json()
            })
            .then(() => {
                this.context.deleteEntry(entryid)
            })
            .catch(error => {
                console.error({error})
            })
    
        
    }
    
    render () {
        const {entry=[]} = this.context
        const entries = entry
        //console.log(entries)
        return (
            entries.map((entryDetail, index) => {
                //console.log(entryDetail)
                return (
                    <div className="journalEntry">
                        <h1>
                            <Link to={`/entry/${entryDetail.entryid}`}>
                                {entryDetail.title}
                            </Link>
                        </h1>
                        <p>{entryDetail.date}</p>
                        <button className='journalEntry_delete' type='button' id={entryDetail.entryid} onClick={this.handleClickDelete}>
                            {' '}
                            Remove
                        </button>
                    </div>
                )
            })
        )
    }
}

export default JournalEntry