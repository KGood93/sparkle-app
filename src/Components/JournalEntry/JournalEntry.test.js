import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import JournalEntry from './JournalEntry';

describe('JournalEntry Component', () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<JournalEntry />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});