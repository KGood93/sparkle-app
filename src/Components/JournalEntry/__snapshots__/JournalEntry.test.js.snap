// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntry Component renders without crashing 1`] = `null`;
